Source/Applications/SystemCenterNotification/Controllers/EmailTypeController.cs

```diff
@@ -264,7 +266,73 @@ public Settings(Action<object> configure) =>
             public APIConfiguration APISettings { get; } = new APIConfiguration();
         }
 
-      
+        public override IHttpActionResult Patch(ScheduledEmailType record)
+        {
+            if (!PatchAuthCheck())
+                return Unauthorized();
+
+            IHttpActionResult result = base.Patch(record);
+            ReconfigureScheduledEmailNode();
+
+            return result;
+        }
+
+        public override IHttpActionResult Post(JObject record)
+        {
+            if (!PostAuthCheck())
+                return Unauthorized();
+
+            IHttpActionResult result = base.Post(record);
+            ReconfigureScheduledEmailNode();
+
+            return result;
+        }
+
+        public override IHttpActionResult Delete(ScheduledEmailType record)
+        {
+            if (!DeleteAuthCheck())
+                return Unauthorized();
+
+            IHttpActionResult result = base.Delete(record);
+            ReconfigureScheduledEmailNode();
+
+            return result;
+        }
+
+        private void ReconfigureScheduledEmailNode()
+        {
+            Settings settings = new Settings(new ConfigurationLoader(CreateDbConnection).Configure);
+            void ConfigureRequest(HttpRequestMessage request)
+            {
+                request.Method = HttpMethod.Get;
+            }
+
+            using (AdoDataConnection connection = CreateDbConnection())
+            {
+                DataTable hosts = connection
+                    .RetrieveData(@"
+                        SELECT ActiveHost.* , Node.ID as NodeID
+                        FROM 
+	                        ActiveHost LEFT JOIN 
+	                        Node ON ActiveHost.ID = Node.HostRegistrationID LEFT JOIN
```

Comment:
We don't need outer joins here since the `WHERE` clause depends on `NodeType.Name` anyway.
```suggestion
	                        ActiveHost JOIN 
	                        Node ON ActiveHost.ID = Node.HostRegistrationID JOIN
```